src/python_testing/matter_testing_infrastructure/generate_data_model_xmls_gni.py

```diff
@@ -52,43 +43,54 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+# This file is generated by
+# src/python_testing/matter_testing_infrastructure/generate_data_model_xmls_gni.py
+#
+# Manually re-generate this file on any changes to the data_model directory.
 
 import("//build_overrides/chip.gni")
+"""
+
+GNI_TEMPLATE = """\
 
-data_model_XMLS = [
+data_model_XMLS_{{ dir }} = [
 {% for name in file_list %}
-    "{{ name }}",
+  "{{ name }}",
 {% endfor %}
 ]
-"""
 
-# Function to find and collect all .xml and .json files
+"""
 
 
-def get_data_model_file_names():
+def get_data_model_file_names(directory: str):
+    ''' Function to find and collect all .xml files'''
     file_list = []
-    for directory in directories:
-        for root, _, files in os.walk(directory):
-            for file in files:
-                if file.endswith(".xml") or file.endswith(".json"):
-                    # Replace absolute path with `${chip_root}` for GNI compatibility
-                    relative_path = os.path.join("${chip_root}", os.path.relpath(root, chip_root), file)
-                    file_list.append(relative_path)
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".xml"):
+                # Replace absolute path with `${chip_root}` for GNI compatibility
+                relative_path = os.path.join("${chip_root}", os.path.relpath(root, chip_root), file)
+                file_list.append(relative_path)
     # Sort files alphabetically
     file_list.sort()
     return file_list
 
-# Main function to generate the data_model_xmls.gni file
-
 
 def generate_gni_file():
-    # Step 1: Find all files and create the sorted file list
-    file_list = get_data_model_file_names()
-
-    # Step 2: Render the template with the file list
+    '''Main function to generate the data_model_xmls.gni file'''
     environment = jinja2.Environment(trim_blocks=True, lstrip_blocks=True)
     template = environment.from_string(GNI_TEMPLATE)
-    output_content = template.render(file_list=file_list)
+    output_content_per_dir = []
+
+    # Step 1: Find all files and create the sorted file list
+    dirs = sorted([d for d in os.listdir(data_model_dir, ) if os.path.isdir(os.path.join(data_model_dir, d))])
```

Comment:
nit, should it just be `os.listdir(data_model_dir)` instead of `os.listdir(data_model_dir, )`?

Reply:
yuppers - changed.